core/storage/btree.rs

```diff
@@ -35,6 +35,25 @@ const BTREE_HEADER_OFFSET_RIGHTMOST: usize = 8; /* if internalnode, pointer righ
 */
 pub const BTCURSOR_MAX_DEPTH: usize = 20;
 
+/// Evaluate a Result<CursorResult<T>>, if IO return IO.
+macro_rules! io {
```

Comment:
Same here what about calling this `return_on_io`. I don't love the idea of seeing code like `io!(something)` and it returning. My main complain is that control flow with macros is hidden and naming is the only way to know without being scared.

Reply:
agreed, see above